src/helpers/proposalStrategiesValue.ts

```diff
@@ -0,0 +1,70 @@
+// import { capture } from '@snapshot-labs/snapshot-sentry';
+import snapshot from '@snapshot-labs/snapshot.js';
+import { getVpValueByStrategy } from './entityValue';
+import db from './mysql';
+import { CB } from '../constants';
+
+type Proposal = {
+  id: string;
+  network: string;
+  start: number;
+  strategies: any[];
+};
+
+const REFRESH_INTERVAL = 60 * 1000;
+const BATCH_SIZE = 100;
+
+async function getProposals(): Promise<Proposal[]> {
+  const query =
+    'SELECT id, network, start, strategies FROM proposals WHERE cb = ? AND start < UNIX_TIMESTAMP() ORDER BY created DESC LIMIT ?';
+  const proposals = await db.queryAsync(query, [CB.PENDING_SYNC, BATCH_SIZE]);
+
+  return proposals.map((p: any) => {
+    p.strategies = JSON.parse(p.strategies);
+    return p;
+  });
+}
+
+async function refreshProposalsVpValues(proposals: Proposal[]) {
+  const query: string[] = [];
+  const params: any[] = [];
+
+  for (const proposal of proposals) {
+    await buildQuery(proposal, query, params);
+  }
+
+  if (query.length) {
+    await db.queryAsync(query.join(';'), params);
```

Comment:
This is confusing, I would expect a function like `buildQuery` to return the `query` var instead of instantiate the variable in the `refreshProposalsVpValues` function to let it updated in `buildQuery`? It feel like with this approach depending on how we call the function it could conflict